packages/core/src/ceramic.ts

```diff
@@ -621,12 +621,29 @@ export class Ceramic implements CeramicApi {
       commit,
       opts as CreateOpts
     )
-    return streamFromState<T>(
+
+    const stream = streamFromState<T>(
       this.context,
       this._streamHandlers,
       state$.value,
       this.repository.updates$
     )
+
+    // add stream to MID indexing if model is present
+    if (stream.metadata.model) {
+      const last_anchor_ts = state$.state.anchorProof
```

Comment:
Looking at this, does it make sense to move the whole block into the helper function - that way it can be called externally to add streams directly if necessary.